### PR TITLE
Add libhomfly

### PR DESCRIPTION
**Pull request by isuruf**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge/help-c-cpp, ready for review. Ignore the linter for now

**Comment by isuruf** [comment]
cc @conda-forge/staged-recipes

File: recipes/libhomfly/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    - bdw-gc
+
+test:
+  commands:
+    - test -f $PREFIX/lib/libhomfly${SHLIB_EXT}
+    - test -f $PREFIX/include/homfly.h
+
+about:
+  home: https://github.com/miguelmarco/libhomfly
+  license: Unlicense
```
Comment (ocefpaf) [inline recipes/libhomfly/meta.yaml]: Unlicensed but the family is MIT and it has a license file? I'm confused :smile:

File: recipes/libhomfly/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  license_family: MIT
```
Comment (ocefpaf) [inline recipes/libhomfly/meta.yaml]: Should the family be Public Domain?

Comment (isuruf) [reply recipes/libhomfly/meta.yaml]: Unlicense is a license. It's basically public domain in places where that's legal and in places like Germany where giving up copyright is not possible, it's like MIT.

**Comment by isuruf** [comment]
Thanks @chrisburr, @ocefpaf

File: recipes/libhomfly/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/libhomfly/meta.yaml]: Yeah. I saw that right after I comment on Wikipedia but it looks like Public Domain would be a better fit for "family." Either way, I don't really care about the family metadata, it is flaky by definition.

**Pull request merged**